app/questionnaire/questionnaire_manager.py

```diff
@@ -184,78 +155,69 @@ def get_first_location(self):
     def _get_first_block(self):
         return self._schema.groups[0].blocks[0].id
 
-    def _previous(self):
-        return self._current.previous.item_id
-
     def get_answers(self):
-        '''
-        This method walks the entire list collecting all answers and as such should
-        only be used for the summary node and submission of data. Otherwise use the
-        more efficient find_answer(id) method
-        :return:
-        '''
-
         # walk the list and collect all the answers
         answers_dict = {}
-
         node = self._first
-        answers = []
+
         while node:
-            node_answers = node.state.get_answers()
-            answers.extend(node_answers)
+            answers_dict.update(node.answers)
             node = node.next
-
-        for answer in answers:
-            if not answer.other:
-                answers_dict[answer.id] = answer.value
-            elif answer.other and isinstance(answer.value, list):
-                answer_values = answer.value[:]
-                if 'other' in answer_values:
```

Comment:
I pulled it out as it was using state on the find answer, he is rewriting it now